deerlab/utils/gof.py

```diff
@@ -52,7 +53,10 @@ def goodness_of_fit(x,xfit,Ndof,sigma='auto'):
     chi2red = 1/Ndof*np.linalg.norm(x - xfit)**2/sigma**2
 
     # R-squared test
-    R2 = 1 - np.sum((x-xfit)**2)/np.sum((xfit-np.mean(xfit))**2)
+    with warnings.catch_warnings():

```

Comment:
This is a bit of a hack. Under which circumstances does divide-by-zero occur? Can you detect these conditions?

Reply:
It is not. There can actually be cases where the fitted distance distributions are all zeros. It is e.g. a solution of a global fit if one of the datasets has weight=0. There is not a clean way of predicting them and the error just happens because R2 is not defined for x=0. So it makes sense that in those cases the R2 returns `inf`. A less hacky implementation that would do the same would be: 
```python
if np.all(xfit)==0:
     R2 = np.inf
else: 
     R2 = 1 - np.sum((x-xfit)**2)/np.sum((xfit-np.mean(xfit))**2) 
```